quilt3distribute/dataset.py

```diff
@@ -100,14 +107,14 @@ def add_license(self, doc_or_link: Union[str, Path]):
         """
         self.readme.append_readme_standards(license_doc_or_link=doc_or_link)
 
-    def index_on_columns(self, columns: List[str]):
+    def set_index_columns(self, columns: List[str]):
```

Comment:
This tells the system which columns to use for metadata attachment to a file. From the README:

> A note on the bin script parameter and the Python dataset function `set_index_columns`: Using the small example dataframe above as an example, if we provided `["Structure"]` as the list of columns to index. Using the small example dataframe above as an example, if we provided `["Structure"]` as the list of columns to index on in Python (using the bin script this would be `-i "Structure"`). `{"Structure": "lysosome"}` gets added as metadata	on in Python (using the bin script this would be `-i "Structure"`). `{"Structure": "lysosome"}` gets added as metadata for both `2d/1.png` and `3d/1.tiff` files; `{"Structure": "laminb1"}` gets added as metadata for both `2d/2.png` and	for both `2d/1.png` and `3d/1.tiff` files; `{"Structure": "laminb1"}` gets added as metadata for both `2d/2.png` and `3d/2.tiff` files; etc.	`3d/2.tiff` files; etc.

> In short: the columns provided will be used for metadata attachment for every file found.	In short: the values in each column provided will be used for metadata attachment for every file found.

If that makes more sense I can add that section in to this doc string.

Reply:
Gotchya. Yeah the snippet from the README is definitely a little clearer. When I read `index` I start thinking of other things. I wonder if a more direct name for this method is `set_metadata_columns`.

Reply:
Definitely, the original reason why it was named `index_column` was because Quilt indexing the metadata but agreed will add more to the comment and update the name.